doc/tutorials/basic/pn_rx.md

```diff
@@ -0,0 +1,320 @@
+
+
+wind_speed = pn.widgets.FloatSlider(
+    value=5, start=0, end=20, step=1, name="Wind Speed (m/s)"
+)
+efficiency = pn.widgets.FloatInput(value=0.3, start=0.0, end=1.0, name="Efficiency (kW/(m/s))")
+submit = pn.widgets.Button(name="Submit")
+
+calculate_power_fn = pn.rx(calculate_power_generation)(wind_speed=wind_speed, efficiency=efficiency)
+
+
+result_fn = calculate_power_fn.rx.when(submit)
+
+pn.Column(
+    wind_speed, efficiency, submit, pn.pane.Markdown(result_fn)
+).servable()
+```
+
+Try changing some of the inputs and clicking the submit Button. Try again. Notice how the `calculate_power_fn` is only run when we click the submit Button - we used `.when` to achieve this effect.
+
+## Harnessing Throttling for Performance
+
+To prevent excessive updates and ensure smoother performance, you can apply throttling. This limits the rate at which certain actions or events occur, maintaining a balanced user experience:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+
+pn.extension()
+
+
+def calculate_power_generation(wind_speed, efficiency):
+    print("calculate", wind_speed)
+    sleep(1)
+    power_generation = wind_speed * efficiency
+    return f"Wind Speed: {wind_speed} m/s, Efficiency: {efficiency}, Power Generation: {power_generation:.1f} kW"
+
+
+wind_speed = pn.widgets.FloatSlider(
+    value=5, start=0, end=20, step=1, name="Wind Speed (m/s)"
+)
+
+efficiency = 0.3
+
+calculate_power_fn = pn.rx(calculate_power_generation)(wind_speed=wind_speed.param.value_throttled, efficiency=efficiency)
+
+pn.Column(
+    wind_speed, pn.pane.Markdown(calculate_power_fn)
+).servable()
+```
+
+Try dragging the slider. Notice that the `calculate_power_generation` function is only run when you release the mouse.
+
+### Binding to reactive expressions
+
+You might want to break down your reactive expressions into smaller sub reactive expressions
+
+```{pyodide}
+import panel as pn
+
+pn.extension()
+
+def power_generation(wind_speed, efficiency):
+    print(wind_speed, efficiency)
+    return wind_speed * efficiency
+
+def power_generation_text(wind_speed, efficiency, power):
+    return f"Wind Speed: {wind_speed} m/s, Efficiency: {efficiency}, Power Generation: {power:.1f} kW"
+
+def to_html(text):
+    return f"<strong>{text}<strong>"
+
+wind_speed = pn.widgets.FloatSlider(
+    value=5, start=0, end=20, step=1, name="Wind Speed (m/s)"
+)
+
+efficiency = 0.3
+
+b_power_generation = pn.rx(power_generation)(wind_speed=wind_speed, efficiency=efficiency)
+
+b_power_generation_text = pn.rx(power_generation_text)(wind_speed, efficiency, b_power_generation)
+
+pn.Column(
+    wind_speed, pn.pane.Str(b_power_generation), pn.pane.Markdown(b_power_generation_text)
+).servable()
+```
+
+If you look in the terminal you will see that `pn.rx` only reruns your functions once every time you change an input widget. This is a big contrast to `pn.bind` which might rerun the functions multiple times.
+
+## Triggering Side Effects with `watch`
```

Comment:
Please provide your *best practice* version of the example in this section Philipp, taking into account https://github.com/holoviz/param/issues/806#issuecomment-1974715821, https://github.com/holoviz/panel/issues/6426 and https://github.com/holoviz/param/issues/912.